### PR TITLE
Provide spec file for rpmbuild

### PR DESCRIPTION
**Pull request by jschmidb**
Provide new spec file for rpmbuild and change install location
of libzpc.pc to libdir.

Signed-off-by: Joerg Schmidbauer <jschmidb@de.ibm.com>

### CODE REVIEW

**Review by sharkcz — CHANGES_REQUESTED** [review]

File: libzpc.spec [hunk 1]
```diff
@@ -0,0 +1,91 @@
+Name:		libzpc
+Version:	1.0.0
+Release:	1%{?dist}
+Summary:	Open Source library for the IBM Z Protected-key crypto feature
+
+Group:		Libraries/Crypto
```
[the remark this answers is not in the document]
Comment (jschmidb) [reply libzpc.spec]: Done.

File: libzpc.spec [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+License:	MIT License
+Url:		https://github.com/opencryptoki/libzpc
+Source0:	%{url}/archive/v%{version}/%{name}-%{version}.tar.gz
+
+ExclusiveArch:	s390x
+BuildRequires:	cmake
```
[the remark this answers is not in the document]
Comment (jschmidb) [reply libzpc.spec]: I added gcc g++ make.
But when I rebuild with mock, I get: 
```
CMake Error at CMakeLists.txt:72 (find_library):
  Could not find JSON_C using the following names: json-c
```
json-c is only needed for building the test programs. The rpm build runs fine.

File: libzpc.spec [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+The IBM Z Protected-key Crypto library libzpc is an open-source library
+targeting the 64-bit Linux on IBM Z (s390x) platform. It provides interfaces
+for cryptographic primitives. The underlying implementations make use of
+z/Architecture's extensive performance-boosting hardware support and its
+protected-key feature which ensures that key material is never present in
+main memory at any time.
+
+%package	devel
+Summary:	Development files for %{name}
+Group:		Libraries/Crypto
```
[the remark this answers is not in the document]
Comment (jschmidb) [reply libzpc.spec]: Done.

File: libzpc.spec [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+#   -DCMAKE_INSTALL_PREFIX=<path> : 
+#        Change the install prefix from `/usr/local/` to `<path>`.
+#   -DCMAKE_BUILD_TYPE=<type> : Choose predefined build options. 
+#        The choices for `<type>` are `Debug`, `Release`, `RelWithDebInfo`, 
+#        and `MinSizeRel`.
+#   -DBUILD_SHARED_LIBS=ON : Build a shared object (instead of an archive).
+#   -DBUILD_TEST=ON : Build the test program.
+#   -DBUILD_DOC=ON : Build the html and latex doc.
+%build
+%cmake -DBUILD_SHARED_LIBS=ON -DCMAKE_BUILD_TYPE=Release
```
[the remark this answers is not in the document]
Comment (jschmidb) [reply libzpc.spec]: OK, removed BUILD_SHARED_LIBS=ON and changed to RelWithDebInfo.

File: libzpc.spec [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+%install
+%cmake_install
+
+
+%check
+%ctest
+
+
+%post -p /sbin/ldconfig
```
[the remark this answers is not in the document]
Comment (jschmidb) [reply libzpc.spec]: Done.

File: libzpc.spec [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+
+%postun -p /sbin/ldconfig
```
[the remark this answers is not in the document]
Comment (jschmidb) [reply libzpc.spec]: Done.

File: libzpc.spec [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+
+%files
+%doc LICENSE README.md CHANGES.md
+%{_libdir}/*
```
[the remark this answers is not in the document]
Comment (jschmidb) [reply libzpc.spec]: Done.

File: libzpc.spec [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+%check
+%ctest
+
+
+%post -p /sbin/ldconfig
+
+%postun -p /sbin/ldconfig
+
+%files
+%doc LICENSE README.md CHANGES.md
```
[the remark this answers is not in the document]
Comment (jschmidb) [reply libzpc.spec]: Done.

File: libzpc.spec [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+Version:	1.0.0
+Release:	1%{?dist}
+Summary:	Open Source library for the IBM Z Protected-key crypto feature
+
+License:	MIT License
+Url:		https://github.com/opencryptoki/libzpc
+Source0:	%{url}/archive/v%{version}/%{name}-%{version}.tar.gz
+
+ExclusiveArch:	s390x
+BuildRequires:	cmake gcc g++ make
```
Comment (sharkcz) [inline libzpc.spec:11]: please put each BR on its own line, it's easier to track changes

File: libzpc.spec [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+#   -DCMAKE_INSTALL_PREFIX=<path> : 
+#        Change the install prefix from `/usr/local/` to `<path>`.
+#   -DCMAKE_BUILD_TYPE=<type> : Choose predefined build options. 
+#        The choices for `<type>` are `Debug`, `Release`, `RelWithDebInfo`, 
+#        and `MinSizeRel`.
+#   -DBUILD_SHARED_LIBS=ON : Build a shared object (instead of an archive).
+#   -DBUILD_TEST=ON : Build the test program.
+#   -DBUILD_DOC=ON : Build the html and latex doc.
+%build
+%cmake -DCMAKE_BUILD_TYPE=RelWithDebInfo
```
Comment (sharkcz) [inline libzpc.spec:58]: I think `RelWithDebInfo` is the system-wide default for `cmake`, can you check if the output will be the same when it's removed? The binary and debug rpms should be the same size.

**Review by sharkcz — CHANGES_REQUESTED** [review]

File: libzpc.spec (hunk 9 above)
Comment (jschmidb) [reply libzpc.spec:11]: Done.
And I added
```
BuildRequires:	json-c-devel
BuildRequires:	glibc-devel
```
The lib also needs libpthread, which is in glibc-devel.

Comment (sharkcz) [reply libzpc.spec]: The compiler brings in its C library, so BR: glibc-devel is redundant, please remove.

File: libzpc.spec (hunk 10 above)
Comment (sharkcz) [reply libzpc.spec]: Sounds like the system-wide setting is OK.

File: libzpc.spec (hunk 9 above)
Comment (jschmidb) [reply libzpc.spec]: Done.

**Pull request merged**